lib/pavilion/dir_db.py

```diff
@@ -92,24 +97,73 @@ def create_id_dir(id_dir: Path) -> (int, Path):
 
         return next_id, next_id_path
 
-
 def default_filter(_: Path) -> bool:
     """Pass every path."""
 
     return True
 
+def is_hex(string: str) -> bool:
+    """Determines if passed string is valide hex or not.
+    :param string: A string to check.
+    :return bool: True if hex, False, if not hex.
+    """
+
+    try:
+        int(string, 16)
+    except ValueError as err:
+        return False
+
+    return True
 
 def select(id_dir: Path,
-           filter_func: Callable[[Path], bool] = default_filter) -> List[Path]:
+           filter_func: Callable[[Path], bool] = default_filter,
+           builds_dir: bool = False) -> List[Path]:
     """Return a list of all test paths in the given id_dir that pass the
     given filter function. The paths returned are guaranteed (within limits)
     to be an id directory, and only paths that pass the filter function
     are returned."""
 
     passed = []
     for path in id_dir.iterdir():
-        if path.name.isdigit() and path.is_dir():
-            if filter_func(path):
-                passed.append(path)
+        if not builds_dir:
+            if path.name.isdigit() and path.is_dir():
+                if filter_func(path):
+                    passed.append(path)
+        else:
+            if is_hex(path.name) and path.is_dir():
+                if filter_func(path):
+                    passed.append(path)
```

Comment:
```python
fn_base = 16 if hex_fn else 10
for path in id_dir.iterdir():
  try:
    int(path.name, fn_base)
  except ValueError:
    continue
  
  # Then you only have to check that it's a directory and the filter function
```

Reply:
In fact, you could just make fn_base the function argument instead of build_dir (and default it to 10).